vent/io/hal.py

```diff
@@ -107,23 +110,48 @@ def secondary_pressure(self) -> float:
             raise RuntimeWarning('Secondary pressure sensor not instantiated. Check your "devices.ini" file.')
 
     @property
-    def flow(self) -> float:
-        """ The measured flow rate.
+    def flow_in(self) -> float:
+        """ The measured flow rate inspiratory side.
+        """
+        self._flow_sensor_in.update()
+        return self._flow_sensor_in.get()

```

Comment:
FYI the hardware on prototype only has one flow sensor, and it's on the expiratory side (although we could reconfigure to inspiratory flow measurement). Do to global supply issues, we won't be able to add a second until we sort out making our own (WIP). 

You may have to pick one for the time being, or we can try and rough out and return an estimate maybe?

Reply:
Are the flow sensors in the HAL interface-agnostic? The sensiron sensor we have on each prototype uses I2C to get data, but the second sensor would derive data from a differential pressure reading taken from a D-lite spirometer, similar to the Princeton physics monitoring project. Everything else looks okay to me.

Reply:
Yes. Setting up that kind of flow sensor would involve making changes to vent/io/config/devices.ini to define the flow sensor as an AnalogSensor and specifying the calibration parameters. 

You would also need to have a channel available on the ADC, or if not then add another ADC. Adding another ADC might be slightly more involved but nothing terrible.

Reply:
It's not important for control at all, only for estimation of volume. Estimating this with one sensor might be a bit more challenging. Can we assume that the propValve setting is the flow on the inspiratory side?

Reply:
No, not really; we could probably estimate it knowing both the control valve setpoint & system pressure though, given: Qin = f(CVstpnt, Pforward, Pback) and assuming backpressure is a nice constant 50 psig